assets/js/utils/rtcUtils.ts

```diff
@@ -62,44 +38,61 @@ export const connectWebrtc = async (
       onConnectionError: () => {
         onError("Error while connecting to WebRTC");
       },
-      onJoinSuccess: (_peerId, _peersInRoom) => {
-        localStream.getTracks().forEach((track) => webrtc.addTrack(track, localStream, {}));
-      },
+      onJoinSuccess: (_peerId, _peersInRoom) => {},
       onJoinError: () => {
         onError("Error while joining WebRTC connection");
       },
-      onTrackReady: ({ stream, peer }) => {
-        if (stream != null) addStream(stream, peer.metadata.displayName);
+      onTrackReady: ({ track, stream, peer }) => {
+        addPresenterMediaStream(
+          peer.metadata.displayName,
+          stream!,
+          track!.kind as SourceType,
+          streamsAvailable,
+          setStreamsAvailable
+        );
       },
       onTrackAdded: (_ctx) => {},
       onTrackRemoved: ({ stream, peer }) => {
         if (stream != null) removeStream(peer.metadata.displayName);
       },
-      onPeerJoined: (peer) => {},
+      onPeerJoined: (_peer) => {},
       onPeerLeft: (peer) => {
         removeStream(peer.metadata.displayName);
       },
       onPeerUpdated: (_ctx) => {},
       onRemoved: (_reason) => {
-        removeStream(name);
-        localStream.getTracks().forEach((track) => track.stop());
+        removeStream(clientName);
+        presenterStreams[clientName].getTracks().forEach((track) => track.stop());
         onError("You were removed from WebRTC connection");
       },
     },
   });
 
   webrtc.join({
-    displayName: name,
+    displayName: clientName,
   });
 
   webrtcChannel.on("mediaEvent", (event) => {
-    console.log("INCONING EVENT", event);
     webrtc.receiveMediaEvent(event.data);
   });
 
   return webrtc;
 };
 
+export const replaceTrack = (
+  webrtc: MembraneWebRTC | null,
+  clientName: string,
+  mediaStream: MediaStream,
+  sourceType: SourceType
+) => {
+  const track = mediaStream.getTracks().find((track) => track.kind == sourceType);
+  if (webrtc != null && track != undefined) {
```

Comment:
Negate this condition and make a return in its body. It will remove one level of nesting.

Reply:
Fixed